cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitorTest.java

```diff
@@ -88,68 +94,67 @@ public void testStateWithOnlyActiveSnapshotWindow() {
   public void testStateWithoutEnoughSnapshotWindows() {
     TestContext context = prepareContext();
     LoadMonitor loadMonitor = context.loadmonitor();
-    MetricSampleAggregator aggregator = context.aggregator();
+    KafkaMetricSampleAggregator aggregator = context.aggregator();
 
     // populate the metrics aggregator.
-    // two samples for each partition except T1P1
-    CruiseControlUnitTestUtils.populateSampleAggregator(2, 4, aggregator, T0P0, 0, SNAPSHOT_WINDOW_MS);
-    CruiseControlUnitTestUtils.populateSampleAggregator(2, 4, aggregator, T0P1, 0, SNAPSHOT_WINDOW_MS);
-    CruiseControlUnitTestUtils.populateSampleAggregator(2, 4, aggregator, T1P0, 0, SNAPSHOT_WINDOW_MS);
-    CruiseControlUnitTestUtils.populateSampleAggregator(2, 1, aggregator, T1P1, 0, SNAPSHOT_WINDOW_MS);
+    // two samples for each partition except T1P1. It has no sample in the first window and one in the second window.
+    CruiseControlUnitTestUtils.populateSampleAggregator(2, 4, aggregator, PE_T0P0, 0, WINDOW_MS, METRIC_DEF);
+    CruiseControlUnitTestUtils.populateSampleAggregator(2, 4, aggregator, PE_T0P1, 0, WINDOW_MS, METRIC_DEF);
+    CruiseControlUnitTestUtils.populateSampleAggregator(2, 4, aggregator, PE_T1P0, 0, WINDOW_MS, METRIC_DEF);
```

Comment:
Did we previously add the following line by mistake?: 
`CruiseControlUnitTestUtils.populateSampleAggregator(2, 1, aggregator, T1P1, 0, SNAPSHOT_WINDOW_MS);`

Reply:
The extrapolation behavior changed a little. So the conditions need to be adjusted accordingly. More specifically, previously if there are less than 1/2 of required samples in a window and `AVG_ADJACENT` does not work, it is considered as `NO_VALID_EXTRAPOLATION`. Now it is `FORCED_INSUFFICIENT`.